File: finalFacebook/Best Time to Buy and Sell Stock II.java
public class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length < 2) {
            return 0;
        }
        int profit = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                profit += prices[i] - prices[i - 1];//each time we meet prices[i] that is larger than last day's prices[i - 1]
            }//we will choose to buy stock on i - 1 day and sell stock on i day
        }
        return profit;
    }
}

File: Python/Array(value)/Best Time to Buy and Sell Stock II.py
# O(N)  O(1)
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit=0
        if len(prices)<=1:
            return 0
        for i in range(1,len(prices)):
            if prices[i]>prices[i-1]:
                maxprofit +=prices[i]-prices[i-1]
        return maxprofit